front/app/components/EsriMap/index.tsx

```diff
@@ -98,144 +80,122 @@ const EsriMap = ({
   const isMobileOrSmaller = useBreakpoint('phone');
   const { data: appConfig } = useAppConfiguration();
 
-  const [map, setMap] = useState<Map | null>(null);
-  const [webMap, setWebMap] = useState<WebMap | null>(null);
+  const [map, setMap] = useState<Map | WebMap | null>(null);
   const [mapView, setMapView] = useState<MapView | null>(null);
   const [referenceLayers, setReferenceLayers] =
     useState<Collection<Layer> | null>(null);
 
   const mapRef = useRef<HTMLDivElement | null>(null);
-  const initialValuesLoaded = useRef(false);
+  const [updateMapViewConfig, setUpdateMapViewConfig] = useState(false);
+  const [initialized, setInitialized] = useState(false);
+
+  const mapRefAvailable = !!mapRef.current;
 
-  // On initial render, create a new map and map view and save them to state variables
   useEffect(() => {
-    if (mapRef.current) {
-      const newMap = new Map();
-      const webMap = webMapId && new WebMap({ portalItem: { id: webMapId } }); // If webMapId is provided, create a Web Map
-
-      const mapView = new MapView({
-        container: mapRef.current, // Reference to DOM node that will contain the view
-        map: webMap || newMap, // Use the Web Map if it exists, or the default Map
-        popupEnabled: false,
-        popup: {
-          dockEnabled: false,
-          dockOptions: {
-            breakpoint: false,
-          },
-        },
-      });
+    if (!mapRefAvailable) return;
```

Comment:
Not sure if this is necessary- maybe `mapRef.current` is always available at this moment. But if it's not, the `useEffect` won't refire if you only reference the `mapRef.current` directly. So that's why I put it in a separate variable